compiler/src/dmd/statementsem.d

```diff
@@ -5054,3 +5054,241 @@ bool checkLabel(GotoStatement gs)
     }
     return false;
 }
+
+/*****************************************
+* Turn an aggregate which is an array into an expression tuple
+* of its elements. I.e., lower
+*     static foreach (x; [1, 2, 3, 4]) { ... }
+* to
+*     static foreach (x; AliasSeq!(1, 2, 3, 4)) { ... }
+*/
+    extern(D) void lowerArrayAggregate(StaticForeach stf, Scope* sc)
+    {
+        auto aggr = stf.aggrfe.aggr;
+        Expression el = new ArrayLengthExp(aggr.loc, aggr);
+        sc = sc.startCTFE();
+        el = el.expressionSemantic(sc);
+        sc = sc.endCTFE();
+        el = el.optimize(WANTvalue);
+        el = el.ctfeInterpret();
+        if (el.op != EXP.int64)
+        {
+            stf.aggrfe.aggr = ErrorExp.get();
+            return;
+        }
+
+        Expressions *es;
+        if (auto ale = aggr.isArrayLiteralExp())
+        {
+            // Directly use the elements of the array for the TupleExp creation
+            es = ale.elements;
+        }
+        else
+        {
+            const length = cast(size_t)el.toInteger();
+            es = new Expressions(length);
+            foreach (i; 0 .. length)
+            {
+                auto index = new IntegerExp(stf.loc, i, Type.tsize_t);
+                auto value = new IndexExp(aggr.loc, aggr, index);
+                (*es)[i] = value;
+            }
+        }
+        stf.aggrfe.aggr = new TupleExp(aggr.loc, es);
+        stf.aggrfe.aggr = stf.aggrfe.aggr.expressionSemantic(sc);
+        stf.aggrfe.aggr = stf.aggrfe.aggr.optimize(WANTvalue);
+        stf.aggrfe.aggr = stf.aggrfe.aggr.ctfeInterpret();
+    }
+
+    /*****************************************
+     * Lower any aggregate that is not an array to an array using a
+     * regular foreach loop within CTFE.  If there are multiple
+     * `static foreach` loop variables, an array of tuples is
+     * generated. In thise case, the field `needExpansion` is set to
+     * true to indicate that the static foreach loop expansion will
+     * need to expand the tuples into multiple variables.
+     *
+     * For example, `static foreach (x; range) { ... }` is lowered to:
+     *
+     *     static foreach (x; {
+     *         typeof({
+     *             foreach (x; range) return x;
+     *         }())[] __res;
+     *         foreach (x; range) __res ~= x;
+     *         return __res;
+     *     }()) { ... }
+     *
+     * Finally, call `lowerArrayAggregate` to turn the produced
+     * array into an expression tuple.
+     *
+     * Params:
+     *     sc = The current scope.
```

Comment:
`src/dmd/statementsem.d(5128): Warning: Ddoc: parameter count mismatch, expected 2, got 1`

Reply:
> `src/dmd/statementsem.d(5128): Warning: Ddoc: parameter count mismatch, expected 2, got 1`

this is the error line: `void lowerNonArrayAggregate(StaticForeach lstf, Scope* sc)`
The arguments are 2 already, Am I missing something? @thewilsonator

Reply:
You mean this ??`
auto nvars = lstf.aggrfe ? lstf.aggrfe.parameters.length : 1;`

Reply:
> `src/dmd/statementsem.d(5128): Warning: Ddoc: parameter count mismatch, expected 2, got 1`

with respect to this do I have to add the second argument like this? @thewilsonator 
* Params:
*     sc = The current scope.
*     lstf = argument of type lowernonarrayaggregate
*/
I did this an error popped up